pkg/kubelet/cm/topologymanager/topology_manager.go

```diff
@@ -89,6 +95,8 @@ type HintProvider interface {
 	// GetPodTopologyHints returns a map of resource names to a list of possible
 	// concrete resource allocations per Pod in terms of NUMA locality hints.
 	GetPodTopologyHints(pod *v1.Pod) map[string][]TopologyHint
+	// AllocatePod is called to trigger the allocation of resources to a pod.
+	AllocatePod(pod *v1.Pod, hint TopologyHint) error
```

Comment:
Adding this API in general makes sense and I'm in favor of extending the API surface to cover a a new usecase rather than shoehorning in the existing API. Plus, this is a natural extension of `GetPodTopologyHints`. So I think this is the right direction, let's iterate to refine as needed.